src/halmos/sevm.py

```diff
@@ -1845,9 +1935,10 @@ def create(
         ex.storage[new_addr] = {}  # existing storage may not be empty and reset here
 
         # transfer value
-        # assume balance is enough; otherwise ignore this path
-        ex.solver.add(UGE(ex.balance_of(caller), value))
         if not (is_bv_value(value) and value.as_long() == 0):
+            # assume balance is enough; otherwise ignore this path
+            ex.solver.add(UGE(ex.balance_of(caller), value))
+
```

Comment:
looks like this significantly duplicates the logic in `send_callvalue()`, maybe we want to factor this out independently? something like `transfer_value(from: Address, to: Address, value: BitVecRef) -> None`

Reply:
done